StatsdClient/Statsd.cs

```diff
@@ -204,13 +234,33 @@ private void InitialiseInternal(Func<IOutputChannel> createOutputChannel, string
         /// <param name="prefix"></param>
         /// <param name="value"></param>
         /// <param name="postFix">A value to append to the end of the line.</param>
-        /// <returns>The formatted metric</returns>
-        protected virtual string PrepareMetric(string metricType, string name, string prefix, string value, string postFix = null)
-        {
-            return (String.IsNullOrEmpty(prefix) ? name : (prefix + "." + name))
-                   + ":" + value
-                   + "|" + metricType
-                   + (postFix == null ? String.Empty : "|" + postFix);
+        /// <returns>The formatted metric</returns>
+        protected virtual string PrepareMetric(string metricType, 
+            string name, string prefix, string value, string namePostfix = null, string postFix = null)
+        {
+            StringBuilder builder = new StringBuilder();
```

Comment:
I've actually experienced the opposite with string concatenations in C# but it's less a speed thing and more about memory since string concatenation will produce additional allocated strings as the result of each individual '+' operation.  Stringbuilder won't do that and I can pass chars to it instead of single-character strings, which further reduces the number of allocations.  Would be simple enough to benchmark though so I'll see if I can compare the two approaches.